src/hooks/useFlag.js

```diff
@@ -1,8 +1,20 @@
-// TODO: Write useFlag hook
+import compareAnswers from "../utils/compareAnswers";
+import { setFlagTrue } from "../redux/missions";
+import { useSelector, useDispatch } from "react-redux";
+import { addRecord } from "./useTerminal";
 
-// checkFlag: (state, { value, missionId, flagId }) => {
-//   // 1. Get mission and flag
-//   // 2. Hash value with md5
-//   // 3. Check if value is correct with saved falg
-//   // 4. If correct, set "done" to true
-// },
+const useFlag = (missionIndex, flagIndex, userAnswer) => {
+  const dispatch = useDispatch();
+  const savedAnswer = useSelector(
+    (state) => state.missions.list[missionIndex].flags[flagIndex].answer
```

Comment:
Tutaj już przydałby się jakiś cusotmowy selector, który dobierze się do odpowiedniego mission po wartości selectedMission, która przecież jest już globalnie przechowywana. Podonbnie w stosunku do currentFlag, ale ta nie powinna być wybierana przez usera, tylko podawana kolejna bez wartości "done". (Tak samo misje wykonane/in-progress możemy (tam gdzie to będzie potrzbne) wyznaczać poprzez sprawedzenie czy wszystkie flagi wewnątrz są "done"